mlflow/langchain/retriever_wrapper.py

```diff
@@ -0,0 +1,127 @@
+"""Chain for wrapping a retriever."""
+from __future__ import annotations
+
+import json
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import yaml
+from langchain.callbacks.manager import AsyncCallbackManagerForChainRun, CallbackManagerForChainRun
+from langchain.chains.base import Chain
+from langchain.schema import BaseRetriever, Document
+from pydantic import Extra, Field
+
+
+class RetrieverWrapper(Chain):
```

Comment:
Can we document this class and also include it in the MLflow docs with examples? Else, it's not obvious how users will construct one of these or why they would